build.rs

```diff
@@ -1,6 +1,30 @@
 use std::env;
 
+include!("src/dist/triple.rs");
+
+pub fn from_build() -> Result<PartialTargetTriple, String> {
+    if let Some(triple) = env::var("RUSTUP_OVERRIDE_BUILD_TRIPLE").ok() {
+        PartialTargetTriple::new(&triple).ok_or(triple)
+    } else {
+        let triple = env::var("TARGET").unwrap();
+        PartialTargetTriple::new(&triple).ok_or(triple)
+    }
```

Comment:
```suggestion
    let triple = env::var("RUSTUP_OVERRIDE_BUILD_TRIPLE").unwrap_or_else(|| env::var("TARGET").unwrap());
    PartialTargetTriple::new(&triple).ok_or(triple)
```

Reply:
While that is fewer lines of code, I'm not sure it's necessarily clearer.  I'll go for deduplicating the call to `::new` but I will use the `if let` rather than the `unwrap_or_else` approach.